cloud-controller-manager/do/resources.go

```diff
@@ -34,62 +35,211 @@ import (
 )
 
 const (
-	controllerSyncPeriod = 1 * time.Minute
-	requestTimeout       = 10 * time.Second
+	controllerSyncTagsPeriod      = 1 * time.Minute
+	controllerSyncResourcesPeriod = 1 * time.Minute
+	requestTimeout                = 1 * time.Minute
+	syncResourcesTimeout          = 3 * time.Minute
```

Comment:
The timeout is currently covering all resources. I wonder if we should define it as timeout per resource type (and reduce accordingly) so that we can add potentially more resource types in the future without having to remember to bump / revisit the timeout?

Reply:
You're right, but I'm not sure how much tweaking this value will need. 3 minutes basically means 3 full-length api requests (I believe timeout is configured at 1 min on DO's api). It's hard to tell as well because depending on the number of resources/pages needed, this could vary wildly. 🤔 

Let's stick with 1 value that large value for now and once we have a better idea (have this running on more clusters/see issues), let's tweak this to be better? Pretty sure we'll need to revisit these timeouts + periods to be based on # of resources in the account.

Reply:
👌